12-project-maintenance.Rmd

```diff
@@ -8,3 +8,121 @@ ottrpal::set_knitr_image_path()
 ```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_58")
 ```
+
+## Understand best practices for maintaining projects
+
+Many R projects are not really finished, they are either maintained long term or they deprecate. Many folks have many R projects over time which means keeping all your R projects maintained could easily become a time consuming endeavor!
+
+In order to keep maintenance a priority without it using up all your time, we have a few tips to discuss. These tips will help you ease the burden of maintenance and lead to a healthier R project long term.
+
+Building some of the infrastructure we will describe will take a bit more time upfront, but will save you time long term.
+
+1. Have unit testing with `testthat`
+2. Use continuous deployment / continuous integration (CI/CD) principles
+
+## What is Unit Testing?
+
+One of the most comprehensive methods for making sure your code works is building tests. Unit testing can be really straightforward to build using the [`testthat` package](https://testthat.r-lib.org/). `testhat` works with the `usethis` R package so you can start by installing both of those.
+
+```
+install.packages(c("usethis", "testthat"))
+```
+
+Then you can create your first test:
+
+```
+usethis::use_test("new_test")
+```
+
+This will open up a page that looks like this:
+
+```
+test_that("multiplication works", {
+  expect_equal(2 * 2, 4)
+})
+```
+
+From this chapter:
+
+> - A test file holds one or more test_that() tests.
+> - Each test describes what it’s testing: e.g. “multiplication works”.
+> - Each test has one or more expectations: e.g. expect_equal(2 * 2, 4).
```

Comment:
I wonder if we could show what we would do for the custom function example- I didn't really pay attention to it, but it would be nice to fit them together to show a more complicated example

Reply:
That's right. At this point they will have gone through the custom functions chapter so we can put it all together here.